### PR TITLE
Use Environment.IsPrivilegedProcess

### PR DESCRIPTION
**Pull request by iSazonov**
Follow #77355.

### CODE REVIEW

File: src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs [hunk 1]
```diff
@@ -14,11 +15,9 @@ namespace System.IO
     /// <summary>Base class for test classes the use temporary files that need to be cleaned up.</summary>
     public abstract partial class FileCleanupTestBase : IDisposable
     {
-        private static readonly Lazy<bool> s_isElevated = new Lazy<bool>(() => AdminHelpers.IsProcessElevated());
-
         private string fallbackGuid = Guid.NewGuid().ToString("N").Substring(0, 10);
 
-        protected static bool IsProcessElevated => s_isElevated.Value;
+        protected static bool IsProcessElevated => Environment.IsPrivilegedProcess;
```
Comment (teo-tsirpanis) [inline src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs:20]: This file is being used by projects that target earlier framework versions and cannot use `Environment.IsPrivilegedProcess`. What you could do is modify `AdminHelpers.IsProcessElevated to use it, and maybe cache the value for the earlier frameworks there.

Comment (iSazonov) [reply src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs:20]: AdminHelpers.IsProcessElevated is used to test Environment.IsPrivilegedProcess

Comment (teo-tsirpanis) [reply src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs:20]: Hmm, so I guess you can't change it here.

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs [hunk 2]
```diff
@@ -9,8 +9,7 @@ namespace BasicEventSourceTests
     public partial class TestEventCounter
     {
         // Specifies whether the process is elevated or not.
-        private static readonly Lazy<bool> s_isElevated = new Lazy<bool>(AdminHelpers.IsProcessElevated);
-        private static bool IsProcessElevated => s_isElevated.Value;
+        private static bool IsProcessElevated => Environment.IsPrivilegedProcess;
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs:12]: I think there's a corresponding member on PlatformDetection, in which case case all of these IsProcessElevated helpers could be deleted, and the ConditionalFact/Theory attributes changed to use that (which is the primary purpose of PlatformDetection).  That consolidation could be done in a separate PR, though.

Comment (iSazonov) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs:12]: Eh, I won't promise to do it later. I use VS Code (mostly for PowerShell dev) and it does not show dependencies in .Net sources. Maybe it's better in VS, but I haven't tried it.

File: src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Unix.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
```
Comment (stephentoub) [inline src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Unix.cs:4]: Why is this needed?

File: src/libraries/System.ServiceProcess.ServiceController/tests/TestServiceProvider.cs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
```
Comment (stephentoub) [inline src/libraries/System.ServiceProcess.ServiceController/tests/TestServiceProvider.cs:4]: Why is this needed?

File: src/libraries/System.Diagnostics.Process/tests/ProcessTests.Unix.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/tests/ProcessTests.Unix.cs:4]: Why is this needed?

**Review by stephentoub — COMMENTED** [review]
Other than my comments, LGTM.

**Review by stephentoub — APPROVED** [review]
Thanks.

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs:12]: https://github.com/dotnet/runtime/blob/da1afc8c45afa7a980e5be450cfb1c65dc80f4d7/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs#L58
https://github.com/dotnet/runtime/blob/da1afc8c45afa7a980e5be450cfb1c65dc80f4d7/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Windows.cs#L256

Comment (jeffhandley) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.Etw.cs:12]: PR created: #78793
